### DISCUSSION TITLE
Multiple query fields are processed sequentially?

### DISCUSSION THREAD

**Discussion by sanchit-gupta-cn** [mirumee/ariadne#520] [Q&A] [answered]
Hi everyone,

According to the GraphQL spec, multiple query fields are processed in parallel by a GraphQL server ([Ref](https://graphql.org/learn/queries/#multiple-fields-in-mutations)), but when working with ariadne, I observed that the fields in a single query operation were being executed sequentially (I am using Django along with `ariadne.wsgi.GraphQL` to serve my GraphQL server).

Is this the intended behavior, or am I missing something? Would be great if someone could help answer my doubt.

PS: Sorry if this seems like a trivial doubt. I am relatively new to GraphQL and ariadne.

**Comment by rafalp** [comment, marked as answer]
GraphQL spec is written with one eye towards the `graphql.js` which is reference implementation that supports asynchronous query execution.

Python counterpart provides two query execution strategies: async and sync. This is because async is opt-in in Python projects. `wsgi.GraphQ` (as well as `ariadne.contrib.django.views.GraphQLView`) are using synchronous strategy because neither `WSGI` or `Django` are fully async stacks.

Parallel execution holds true for `ariadne.asgi.GraphQL`.

**Reply by sanchit-gupta-cn** [reply]
Hi @rafalp, thanks a lot for the explanation. We'll re-evaluate our stack and see if it is worth it to shift to an async one.
